### PR TITLE
Header, footer, routing

### PR DESCRIPTION
**Pull request by viraxslot**
ToDo:
- [x] Configured husky, prettier and lint-staged
- [x] Added pull request pipeline checks 
- [x] Add Header and Footer components
- [x] Add router with basic routes

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -2,10 +2,21 @@
...
+      href="//code.ionicframework.com/ionicons/2.0.1/css/ionicons.min.css"
+      rel="stylesheet"
+      type="text/css"
+    />
+    <link
+      href="//fonts.googleapis.com/css?family=Titillium+Web:700|Source+Serif+Pro:400,700|Merriweather+Sans:400,700|Source+Sans+Pro:400,300,600,700,300italic,400italic,600italic,700italic"
+      rel="stylesheet"
+      type="text/css"
+    />
+    <link rel="stylesheet" href="//demo.productionready.io/main.css" />
```
Comment (MarkLauer) [inline index.html:17]: we can leave these links as they are for now, but usually it's better to import styles and fonts in your code directly, so the app is not dependent on any third-party resources

File: src/components/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          conduit
+        </a>
+        <ul className="nav navbar-nav pull-xs-right">
+          {pages.map((page: PageName, idx: number) => {
+            return (
+              <li className="nav-item" key={idx}>
+                <NavLink
+                  to={Paths[page]}
+                  className={({ isActive, isPending }) =>
+                    `nav-link ${isPending ? 'pending' : isActive ? 'active' : ''}`
```
Comment (MarkLauer) [inline src/components/Header.tsx:20]: there's a cool library to handle this type of situations :) it's not mandatory of course, but working with classnames will be much more convenient with it, give it a try if you want
https://github.com/lukeed/clsx

File: src/pages/SignUpPage.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Footer } from '../components/Footer';
+import { Header } from '../components/Header';
+
+export default function SignUpPage() {
+  return (
+    <div id="sign-up-page">
+      <Header />
+      <p>Sign up page placeholder</p>
+      <Footer />
```
Comment (MarkLauer) [inline src/pages/SignUpPage.tsx:9]: all your pages duplicate rendering of header and footer, which not only inconvenient, but also could lead to bugs like losing local state of header or footer after navigation, please use nested routes to render them instead
https://reactrouter.com/en/main/start/tutorial#nested-routes

File: .eslintrc.cjs [hunk 4]
```diff
@@ -5,14 +5,12 @@ module.exports = {
...
   ],
   ignorePatterns: ['dist', '.eslintrc.cjs'],
   parser: '@typescript-eslint/parser',
   plugins: ['react-refresh'],
   rules: {
-    'react-refresh/only-export-components': [
-      'warn',
-      { allowConstantExport: true },
-    ],
+    'react-refresh/only-export-components': ['warn', { allowConstantExport: true }],
```
Comment (MarkLauer) [inline .eslintrc.cjs:14]: is it a default rule from vite template?

Comment (viraxslot) [reply .eslintrc.cjs:14]: Yeap, should I delete it?

File: index.html (hunk 1 above)
Comment (viraxslot) [reply index.html]: Fixed, hopefully in a right way

File: src/components/Header.tsx (hunk 2 above)
Comment (viraxslot) [reply src/components/Header.tsx]: Fixed using the library

File: src/pages/SignUpPage.tsx (hunk 3 above)
Comment (viraxslot) [reply src/pages/SignUpPage.tsx:9]: Let's discuss, because I fix in the other way in the next PR

Comment (viraxslot) [reply src/pages/SignUpPage.tsx:9]: Will fix in the next PR

File: src/components/Footer.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+export function Footer() {
+  return (
+    <footer>
+      <div className="container">
+        <a href="/" className="logo-font">
```
Comment (MarkLauer) [inline src/components/Footer.tsx:5]: please use react-router link here https://reactrouter.com/en/main/components/link

Comment (viraxslot) [reply src/components/Footer.tsx]: Fixed

**Review by MarkLauer — APPROVED** [review]

**Pull request merged**